### PR TITLE
Apply 1.x changes to 2.x

### PR DESCRIPTION
**Pull request by KovalevAndrey**
## Description

Applying latest 1.x changes to 2.x

## Check list

- [x] I have updated `CHANGELOG.md` if required.
- [x] I have updated documentation if required.

### CODE REVIEW

File: libraries/core/src/test/kotlin/com/bumble/appyx/core/lifecycle/MinimumCombinedLifecycleTest.kt [hunk 1]
```diff
@@ -39,10 +39,10 @@ class MinimumCombinedLifecycleTest {
 
     @Test
     fun `destroyed when one of lifecycles is destroyed`() {
-        val test1 = TestLifecycle()
-        test1.state = State.DESTROYED
-        val test2 = TestLifecycle()
+        val test1 = TestLifecycle().apply { state = State.RESUMED }
```
Comment (CherryPerry) [inline libraries/core/src/test/kotlin/com/bumble/appyx/core/lifecycle/MinimumCombinedLifecycleTest.kt]: Pass into constructor instead.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,132 +2,5 @@
 
 ## Pending changes
 
-- [#217](https://github.com/bumble-tech/appyx/pull/217) – **Updated**: Moved `navigation-compose` sample into `:samples:app`.
```
Comment (CherryPerry) [inline CHANGELOG.md:5]: Keep it?

File: documentation/navigation/model-driven-navigation.md [hunk 3]
```diff
@@ -5,9 +5,9 @@
 
 Generally speaking, most navigation solutions have fixed navigation mechanisms (e.g. a back stack).
 
-Appyx gives you the freedom to define your own navigation model. For example, you can implement any of the examples you see here with with the same approach:
+Appyx gives you the freedom to define your own navigation model. For example, you can implement any of the examples you see here with the same approach:
 
-<img src="https://i.imgur.com/N8rEPrJ.gif" width="150"> <img src="https://i.imgur.com/esLXh61.gif" width="150"> <img src="https://i.imgur.com/8gy3Ghb.gif" width="150"> <img src="https://i.imgur.com/V3bks8c.gif" width="150">
+<img src="https://i.imgur.com/N8rEPrJ.gif" width="150"> <img src="https://i.imgur.com/esLXh61.gif" width="150"> <img src="https://i.imgur.com/8gy3Ghb.gif" width="150"> <img src="https://cdn-images-1.medium.com/max/1600/1*mEg8Ebem3Hd2knQSA0yI1A.gif" width="150">
```
Comment (CherryPerry) [inline documentation/navigation/model-driven-navigation.md:10]: I am not sure how `imgur` is reliable, but feels like `cdn-images-1.medium.com` is definitely not.

**Review by CherryPerry — APPROVED** [review]

File: gradle/libs.versions.toml [hunk 4]
```diff
@@ -4,10 +4,10 @@
 androidCompileSdk = "33"
 androidMinSdk = "21"
 androidTargetSdk = "32"
-androidx-lifecycle = "2.4.1"
+androidx-lifecycle = "2.5.1"
```
Comment (LachlanMcKee) [inline gradle/libs.versions.toml:7]: I don't suppose we could update this on 1.x as well? it's a blocker for adding dagger in a future PR
https://github.com/bumble-tech/appyx/pull/115

File: documentation/navigation/model-driven-navigation.md (hunk 3 above)
Comment (KovalevAndrey) [reply documentation/navigation/model-driven-navigation.md:10]: > I am not sure how `imgur` is reliable, but feels like `cdn-images-1.medium.com` is definitely not.

I think they're both reliable. Github doesn't show files > 10 Mb, it's easy to miss

File: gradle/libs.versions.toml (hunk 4 above)
Comment (KovalevAndrey) [reply gradle/libs.versions.toml:7]: I'll update now in a separate PR

File: libraries/core/src/test/kotlin/com/bumble/appyx/core/lifecycle/MinimumCombinedLifecycleTest.kt [hunk 5]
```diff
@@ -39,10 +39,10 @@ class MinimumCombinedLifecycleTest {
 
     @Test
     fun `destroyed when one of lifecycles is destroyed`() {
-        val test1 = TestLifecycle()
-        test1.state = State.DESTROYED
-        val test2 = TestLifecycle()
+        val test1 = TestLifecycle(State.RESUMED)
```
Comment (KovalevAndrey) [inline libraries/core/src/test/kotlin/com/bumble/appyx/core/lifecycle/MinimumCombinedLifecycleTest.kt:42]: test changes due to behaviour change in https://developer.android.com/jetpack/androidx/releases/lifecycle#2.5.0-alpha03

File: CHANGELOG.md (hunk 2 above)
Comment (KovalevAndrey) [reply CHANGELOG.md:5]: why? these changes relate to 1.0?

**Pull request merged**